src/controllers/addressBook/addressBook.ts

```diff
@@ -0,0 +1,160 @@
+import { getAddress } from 'ethers'
+
+import { Storage } from '../../interfaces/storage'
+import EventEmitter from '../eventEmitter/eventEmitter'
+
+export type Contacts = Array<{
+  name: string
+  address: string
+  isWalletAccount?: boolean
+}>
+
+/**
+ * Address Book controller
+ */
+
+export class AddressBookController extends EventEmitter {
+  // Manually added contact (stored in storage)
+  #manuallyAddedContacts: Contacts = []
+
+  // Contacts, generated on the fly from the accounts in the wallet (not stored in storage)
+  #accountsInWalletContacts: Contacts = []
```

Comment:
I tend to like long descriptive names, for this scenario, I wonder if these names are better (they could not be, since this is opinionated):

```
#contactsSourcedFromAccounts
#contactsManuallyAdded
```